test/jit/test_class_type.py

```diff
@@ -1007,3 +1007,23 @@ def forward(self):
                 return new_obj.a
 
         self.checkModule(ModuleWithMeta(5), ())
+
+    def test_type_annotation(self):
+        """
+        Test that annotating container attributes with types works correctly
+        """
+        @torch.jit.script
+        class CompetitiveLinkingTokenReplacementUtils:
+            def __init__(self):
+                self.my_list : List[Tuple[float, int, int]] = []
+                self.my_dict : Dict[int, int] = {}
+
+        @torch.jit.script
+        def foo():
+            y = CompetitiveLinkingTokenReplacementUtils()
+            new_dict : Dict[int, int] = {1: 1, 2: 2}
+            y.my_dict = new_dict
+
+            new_list : List[Tuple[float, int, int]] = [(1.0, 1, 1)]
+            y.my_list = new_list
+            return y
```

Comment:
Do we need an execution test? I think omitting one is fine because the typechecking for `y.my_dict = new_dict` and `y.my_list = new_list` is enough to test that the type annotations for `self.my_list` and `self.my_dict` are not ignored, but I just wanted to make sure.

Reply:
Yeah, this is a test on the static parts of the type system, so if the thing compiles correctly then we are good from the perspective of this unit test